### PR TITLE
Move audio to an audio package and add support for web

### PR DESCRIPTION
**Pull request by otraore**
Work in progress. Also depends on engo.io/audio package which is also a work in progress. 
Status:
Desktop/Native sounds the same (nothing broke as far as I know)
Web sounds also work, but we're making a move to http://www.html5rocks.com/en/tutorials/webaudio/positional_audio/ in the next few days on the audio package

### CODE REVIEW

File: assets.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ func (formats *Formats) SetRoot(root string) {
 	formats.root = root
 }
 
+// GetRoot returns the currently set root directory for assets
+func (formats *Formats) GetRoot() string {
```
Comment (EtienneBruines) [inline assets.go]: Idiomatic Go would name this `Root() string`.

File: common/audio_desktop.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
+//+build !windows,!netgo,!android
```
Comment (EtienneBruines) [inline common/audio_desktop.go]: I guess in the future (as in: before merging), the `engo.io/audio` API would be such that we don't need an `audio_desktop.go` file here, but just one `audio.go` file?

File: demos/hello-world/hello.go [hunk 3]
```diff
@@ -42,7 +42,7 @@ func (*DefaultScene) Setup(w *ecs.World) {
 		Scale:    engo.Point{8, 8},
 	}
 	guy.SpaceComponent = common.SpaceComponent{
-		Position: engo.Point{0, 0},
+		Position: engo.Point{1, 1},
```
Comment (EtienneBruines) [inline demos/hello-world/hello.go]: Why?

Comment (otraore) [inline demos/hello-world/hello.go]: Sorry this was me testing something when we had that vanvasWidth/Height issue.

File: common/audio_desktop.go (hunk 2 above)
Comment (otraore) [inline common/audio_desktop.go]: Hopefully, will be a little tricky though because of openal but I have an idea in mind.

**Comment by Noofbiz** [comment]
Closed by #494

**Pull request closed without merging**